addons/sale/models/sale.py

```diff
@@ -573,9 +584,6 @@ def action_invoice_create(self, grouped=False, final=False):
             # Use additional field helper function (for account extensions)
             for line in invoice.invoice_line_ids:
                 line._set_additional_fields(invoice)
-            # Necessary to force computation of taxes. In account_invoice, they are triggered
-            # by onchanges, which are not triggered when doing a create.
-            invoice.compute_taxes()
```

Comment:
Not sure about this. Actually, I'm not sure e6ff16cbba3802ca17c7bc110d91674580edd38b was the right fix in a first place. IMHO, removing the tax computation after quantities are modified (sign might have changed) and additional fields are set is not very safe. I'd rather have useless processing rather than incorrect tax computation.

Reply:
I was also unsure of this, which is why I did it in a separate commit... I don't know anything about the functional aspect of this so I leave it to yours and @qdp-odoo 's judgement, it just seemed weird to have the recompute twice here, and in fact the original PR did not include the recompute, it was added by @qdp-odoo before merging.

Reply:
OK for the first commit which brings the main improvement, then. For this one, we'll check if we can prevent the computation of taxes twice.